integration-tests/utils/integration-utils.js

```diff
@@ -181,24 +183,53 @@ const delay = ms => {
 }
 
 const dismissNotifications = async page => {
-  await delay(3000) // delayed for any alerts to show
-  const notificationCloseButtons = await page.$x(
-    '(//a | //*[@role="button"] | //button)[contains(@aria-label,"Dismiss") and not(contains(@aria-label,"error"))]')
-
+  const notificationCloseButtonXpath = '(//a | //*[@role="button"] | //button)[contains(@aria-label,"Dismiss") and not(contains(@aria-label,"error"))]'
+  // Don't sleep for 3 seconds if a notification was found already
+  await Promise.race([
+    delay(3000), // delayed for any alerts to show
+    page.waitForXPath(notificationCloseButtonXpath, { visible: true })
+  ])
+  const notificationCloseButtons = await page.$x(notificationCloseButtonXpath)
   await Promise.all(
-    notificationCloseButtons.map(handle => handle.click())
+    notificationCloseButtons.map(button => button.evaluateHandle(btn => btn.click(), button))
```

Comment:
Changed how click works here because `elementHandle.click()` uses element's (x, y) coordinates. If there are more than one notifications, after one notification is closed, position of next notification moves. So click by coordinates is bad idea here as closing additional notifications can unexpectedly fail.

Reply:
It seems like this could be simplified to
```js
notificationCloseButtons.map(button => button.evaluate(btn => btn.click()))
```
`evaluate` and `evaluateHandle` automatically pass the handle as the first argument to the function, so passing `button` again is unnecessary.

https://pptr.dev/#?product=Puppeteer&version=v15.1.1&show=api-elementhandleevaluatepagefunction-args